test/util.spec.js

```diff
@@ -24,150 +23,89 @@ describe('util', () => {
-      expect(cid.multihash).to.exist()
-      const mh = multihash.decode(cid.multihash)
-      expect(mh.name).to.equal('sha2-256')
-      done()
-    })
+  it('.serialize and .deserialize object with slash as property', () => {
+    const slashObject = { '/': true }
+    const serialized = dagCBOR.util.serialize(slashObject)
+    const deserialized = dagCBOR.util.deserialize(serialized)
+    expect(deserialized).to.eql(slashObject)
   })
 
-  it('.cid with hashAlg', (done) => {
-    dagCBOR.util.cid(obj, { hashAlg: 'sha2-512' }, (err, cid) => {
-      expect(err).to.not.exist()
-      expect(cid.version).to.equal(1)
-      expect(cid.codec).to.equal('dag-cbor')
-      expect(cid.multihash).to.exist()
-      const mh = multihash.decode(cid.multihash)
-      expect(mh.name).to.equal('sha2-512')
-      expect(mh.length).to.equal(64)
-      done()
-    })
+  it('error catching', () => {
+    const circlarObj = {}
+    circlarObj.a = circlarObj
+    expect(() => dagCBOR.util.serialize(circlarObj)).to.throw(
+      'The object passed has circular reference')
   })
 
-  it('.cid with hashAlg and hashLen', (done) => {
-    dagCBOR.util.cid(obj, { hashAlg: 'keccak-256', hashLen: 28 }, (err, cid) => {
-      expect(err).to.not.exist()
-      expect(cid.version).to.equal(1)
-      expect(cid.codec).to.equal('dag-cbor')
-      expect(cid.multihash).to.exist()
-      const mh = multihash.decode(cid.multihash)
-      expect(mh.name).to.equal('keccak-256')
-      expect(mh.length).to.equal(28)
-      // The CID must be 32 bytes including 4 bytes for
-      // <cid-version><multicodec><hash-function><digest-size>
-      expect(cid.buffer.length).to.equal(32)
-      expect(cid.toBaseEncodedString()).to.equal('z6dSUELEcAsg5oXs7gsv42rYfczTLizSBTpGUa5M3bxe')
-      done()
-    })
+  it('.cid', async () => {
+    const cid = await dagCBOR.util.cid(serializedObj)
+    expect(cid.version).to.equal(1)
+    expect(cid.codec).to.equal('dag-cbor')
+    expect(cid.multihash).to.exist()
+    const mh = multihash.decode(cid.multihash)
+    expect(mh.name).to.equal('sha2-256')
   })
 
-  it('strings', (done) => {
-    dagCBOR.util.cid('some test string', (err, cid) => {
-      expect(err).to.not.exist()
-      expect(cid.version).to.equal(1)
-      expect(cid.codec).to.equal('dag-cbor')
-      expect(cid.multihash).to.exist()
-      done()
-    })
+  it('.cid with hashAlg', async () => {
+    const cid = await dagCBOR.util.cid(serializedObj, { hashAlg: 'sha2-512' })
+    expect(cid.version).to.equal(1)
+    expect(cid.codec).to.equal('dag-cbor')
+    expect(cid.multihash).to.exist()
+    const mh = multihash.decode(cid.multihash)
+    expect(mh.name).to.equal('sha2-512')
+    expect(mh.length).to.equal(64)
   })
 
-  it.skip('serialize and deserialize - garbage', (done) => {
-    const inputs = []
-
-    for (let i = 0; i < 1000; i++) {
-      inputs.push({ in: garbage(100) })
+  it('fuzz serialize and deserialize with garbage', () => {
```

Comment:
will need to be `.skip`ped right cause this still fails in the browser tests right?

Reply:
Nope it doesn't fail anymore thanks to https://github.com/ipld/js-ipld-dag-cbor/pull/107#pullrequestreview-229914777.

Reply:
oh, that's awesome!